lib/phoenix_live_view/html_algebra.ex

```diff
@@ -176,14 +153,7 @@ defmodule Phoenix.LiveView.HTMLAlgebra do
     {:block, group}
   end
 
-  # Don't format inline tags that are surrounded by texts without whitespaces.
-  # For instance the examples below will not be formatted:
-  #
-  #   text<a class="bar">link</a>text
-  #   <a class="bar">link</a>text
-  #   text<a class="bar">link</a>
-  defp to_algebra({:tag_block, name, attrs, block, %{format?: false}}, context)
-       when name in @inline_elements do
+  defp to_algebra({:tag_block, name, attrs, block, %{preserve?: true}}, context) do
```

Comment:
Maybe instead of `preserve?` we could have `mode`. `mode` would be either `:block`, `:inline`, or `:preserve`. This way you don't need to check `name in HTMLFormatter.inline_elements()` and all relevant information will be handled at the formatter. WDYT?

Reply:
Nice. Just did it :)